### ISSUE TITLE
Server error

### ISSUE THREAD

**Issue by jpsim** [thumbsup/lambda-cloudfront-cookies#2]
First off, huge thanks for putting this together. Looks exactly like what I want to set up. 👌 

However, after following all the steps, I can't seem to get the lambda function to successfully "log in":

```
curl -X POST -d "{\"username\":\"MY_USERNAME\", \"password\":\"MY_PASSWORD\"}" -H "Content-Type: application/json" -i "https://0000000000.execute-api.us-west-1.amazonaws.com/Prod/login" 
HTTP/1.1 500 Internal Server Error
Content-Type: application/json
Content-Length: 12
Connection: keep-alive
Date: Fri, 16 Jun 2017 00:32:27 GMT
x-amzn-RequestId: 4555c638-522b-11e7-b0db-9b020bd73a15
X-Amzn-Trace-Id: sampled=0;root=1-59432719-509f99d0578ae66e8ee848ce
X-Cache: Error from cloudfront
Via: 1.1 81871f1c889cc44b6c25e3ef722a3801.cloudfront.net (CloudFront)
X-Amz-Cf-Id: lMLUkiAOkyh08ry0XXLpHTTPdHGISOzNkU4TDRTBLYELAw1MIId04g==

Server error
```

Would you have any advice as to how to debug this? My AWS-fu is pretty weak 😅 

Thanks again.

**Comment by jpsim** [comment]
For what it's worth, the lambda seems to be (correctly) returning a 403 when I make a GET request, leading me to guess that it's (for the most part) properly configured.

```
curl -H "Content-Type: application/json" -i "https://0000000000.execute-api.us-west-1.amazonaws.com/Prod/login" 
HTTP/1.1 403 Forbidden
Content-Type: application/json
Content-Length: 42
Connection: keep-alive
Date: Fri, 16 Jun 2017 00:45:58 GMT
x-amzn-RequestId: 29a785fa-522d-11e7-ad0d-bda8eb1b4771
x-amzn-ErrorType: MissingAuthenticationTokenException
X-Cache: Error from cloudfront
Via: 1.1 5574699f9d8b8aad21acf2edbb25dc62.cloudfront.net (CloudFront)
X-Amz-Cf-Id: JxQpHV4i7dxg0tGUyklWK3KuPPMXxxljXC7vHevNvMFJ96Vp5B2sdA==

{"message":"Missing Authentication Token"}
```

**Comment by jpsim** [comment]
When I realized that I could add `console.log(...)` calls to `index.js`, redeploy & see the logs in CloudWatch, this became much easier to debug. Hurray for print-driven debugging 🎉 

Turns out that because my KMS key had been created in the `us-east-1` region but everything else was happening in `us-west-1`, I needed to tell `index.js` about it (`const kms = new AWS.KMS({region: 'us-east-1'})`). I also had to specify the `--region us-east-1` option to all the `aws kms encrypt` commands in the instructions, but that was pretty easy to guess.

Finally, I had to add the `LambdaCloudformationCookies` role to the KMS encryption key after running `./deploy` for the lambda to have access to it.

This last step you might want to add to the readme.
